File: Source/Extensions/Tessa.Extensions.Default.Client/Tiles/ExtendedDefaultTypeGroupCaptionsTileExtension.cs
using System.Threading.Tasks;
using Tessa.Extensions.Default.Shared;
using Tessa.UI.Tiles;
using Tessa.UI.Tiles.Extensions;

namespace Tessa.Extensions.Default.Client.Tiles
{
    public sealed class ExtendedDefaultTypeGroupCaptionsTileExtension : TileExtension
    {
        #region Base Overrides

        public override Task InitializingGlobal(ITileGlobalExtensionContext context)
        {
            context.Workspace.SetTypeGroupCaption(DefaultTileNames.Routes, "$KrTiles_Routes");
            return Task.CompletedTask;
        }

        #endregion
    }
}

File: WebClient SDK/src/default/tiles/extendedDefaultTypeGroupCaptionsTileExtension.ts
import { TileExtension, ITileGlobalExtensionContext, ITileWorkspace } from 'tessa/ui/tiles';
import { tryGetFromInfo } from 'tessa/ui';
import { IStorage } from 'tessa/platform/storage';

export class ExtendedDefaultTypeGroupCaptionsTileExtension extends TileExtension {

  public initializingGlobal(context: ITileGlobalExtensionContext) {
    ExtendedDefaultTypeGroupCaptionsTileExtension.setTypeGroupCaption(context.workspace, 'Routes', '$KrTiles_Routes');
  }

  private static setTypeGroupCaption(workspace: ITileWorkspace, group: string, caption: string) {
    let captions = tryGetFromInfo<IStorage | null>(workspace.info, 'TypeGroupCaption', null);
    if (!captions) {
      captions = {};
      workspace.info['TypeGroupCaption'] = captions;
    }

    captions[group] = caption;
  }

}